docs/proposals/service-catalog.md

```diff
@@ -0,0 +1,123 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.1/docs/proposals/service-catalog.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Service Catalog
+
+## Abstract
+
+Kubernetes has Services but they are scoped per namespace, and (assuming eventual support for RBAC) you may not want everyone to be able to see and use all the services in your namespace.
+
+
+This proposal is about:
+
+- associating additional configuration/secret data with Services
+- adding a new Service Catalog resource into which users can publish services that others can easily find and consume
+- adding support for dynamic provisioning of resources from the catalog
+- making it easier to link Services (and their configuration data) to Pods
+
+## Service configuration data
+
+The only data currently associated with a service is a list of zero or more endpoints. Many services also have configuration data associated with them. These include things such as logins, passwords, and additional connection parameters. To tie configuration data to a service, we propose adding information to the Service type to be able to express a relationship between the Service and its relevant configuration data. This could be accomplished by adding new field(s) or by using annotations. Potential configuration data target reference types include Secrets and ConfigData.
```

Comment:
The claim in the future may reference many types of objects - a separate
resource is interesting but ultimately may not help - esp. If the resources
have an existence dependency on the claim.

On Nov 19, 2015, at 9:58 PM, Derek Carr <notifications@github.com> wrote:

In docs/proposals/service-catalog.md
<https://github.com/kubernetes/kubernetes/pull/17543#discussion_r45431467>:

> +
> +## Abstract
> +
> +Kubernetes has Services but they are scoped per namespace, and (assuming eventual support for RBAC) you may not want everyone to be able to see and use all the services in your namespace.
> +
> +
> +This proposal is about:
> +
> +- associating additional configuration/secret data with Services
> +- adding a new Service Catalog resource into which users can publish services that others can easily find and consume
> +- adding support for dynamic provisioning of resources from the catalog
> +- making it easier to link Services (and their configuration data) to Pods
> +
> +## Service configuration data
> +
> +The only data currently associated with a service is a list of zero or more endpoints. Many services also have configuration data associated with them. These include things such as logins, passwords, and additional connection parameters. To tie configuration data to a service, we propose adding information to the Service type to be able to express a relationship between the Service and its relevant configuration data. This could be accomplished by adding new field(s) or by using annotations. Potential configuration data target reference types include Secrets and ConfigData.

I wonder if this should be treated more generally as a 'Package' which
would just hold a list of references to other resources rather than
additional metadata on the service itself? So a "Package" would related the
service with its configuration data, instead of the value being directly on
the service itself.

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/17543/files#r45431467>.